tools/clang/test/SemaHLSL/hlsl/objects/HitObject/hitobject_traceinvoke_payload_udt.hlsl

```diff
@@ -16,7 +16,8 @@ struct Attribs
 [shader("raygeneration")]
 void RayGen()
 {
-  // expected-error@+1{{payload parameter 'payload_in_rg' must be a user-defined type composed of only numeric types}}
+  // expected-error@+2{{object 'dx::HitObject' is not allowed in payload parameters}}
+  // expected-note@8{{'dx::HitObject' field declared here}}
```

Comment:
For this and those below, I have a mild preference that the notes be placed on the lines they reference instead of hard coding the line number. It would be more robust to test additions that way.

Reply:
I don't have a strong opinion on this. Some things i considered when i decided to group the notes with their error:
- more test code may be added below without moving the lines that the notes refer to
- the grouping makes clear what error caused the note to be emitted
- it is trivial to update line numbers even if prior lines need to be changed

Reply:
no objection since.. assuming this is accepted